test/FunctionalTests/Microsoft.OData.Edm.Tests/Csdl/CsdlWriterTests.cs

```diff
@@ -2810,6 +2810,85 @@ public void ShouldSubstituteFullyQualifiedNamespaceWithAliasIfAliasIsSet()
       }
     }
   }
+}");
+        }
+
+        [Fact]
+        public void VerifyKeyAliasForComplexPropertiesIsSupported()

```

Comment:
What about aliases that go through entity navigation properties?

Reply:
done